### ISSUE TITLE
Undo not working properly in Song editor

### ISSUE THREAD

**Issue by jibin1573** [LMMS/lmms#4471]
Using Windows 10 (Version 1804)
LMMS 1.2.0 RC6

I opened a song project, and took the playhead to the start of the song and then pressed Shift+Insert key to add 5 empty bars; then I tried to undo it, but there is no response at all! LMMS is unable to undo this. 

(Also note that, I went through the above mentioned steps directly after opening the project, so there is nothing else to be undine.)

**Comment by jibin1573** [comment]
I have encountered another problem with Undo. Recently, I opened a project and changed the colour of a BB track in the Song editor and tried to undo this, but there is no response.

**Comment by steven-eckhoff** [comment]
I am interested in looking into this. Is anyone else working on it?

**Comment by SecondFlight** [comment]
I don't believe so. It's all yours if you want it! @PhysSong this would be a good thing to base off stable1.2, correct?

**Comment by steven-eckhoff** [comment]
Okay I will take a look then.

Also I was going to ask about an architecture doc for guidance about the expected behavior and I found this [1]. It looks like it is limited at the moment, so I will just ask questions here.

[1] https://github.com/LMMS/lmms/wiki/LMMS-Architecture

**Comment by tresf** [comment]
@Anonymouqs can you please consolidate all "Undo" related bugs into one `meta` issue for us with a checkbox next to each?

@steven-eckhoff feel free to post any updated here or -- once created -- in the `meta` bug report.

**Comment by oldcastlehq** [comment]
I'm not sure if I'm supposed to add anything else here, but the issues with undo in Song Editor still there. Any development here?

**Comment by musikBear** [comment]
@oldcastlehq What issue specific?
And
please post your:
* Name of your **Operative System** (Windows/ Linux/ Mac) And subtype (Home' 'Mint' '10.13' aso)
* lmms **version** and the *site* where you got it from

**Comment by qnebra** [comment]
I guess not all operations in Song Editor are indexed.

**Comment by musikBear** [comment]
> 
> 
> @musikBear no appimage here.

That is wrong! The only linux version that _is_ official, is appImage
It is this one:
https://github.com/LMMS/lmms/releases/download/v1.2.2/lmms-1.2.2-linux-x86_64.AppImage
You should follow the version-update-guide:
https://lmms.io/wiki/index.php?title=Installing_LMMS#Updating_and_installing_new_versions
and upgrade to 1.2.2 Off.
Then retry the bug-issue, however qnebra will already have done that, i hope.. -But for you the off version should replace the one you have now, if you want to have best non-alpha installed.

**Comment by oldcastlehq** [comment]
I think there is a misunderstanding here. 
I'm using 1.3 alpha. This is a compiled version using AUR from Arch Linux https://aur.archlinux.org/packages/lmms-git/.

The issue described here, still there.

**Comment by musikBear** [comment]
> 
> 
> I think there is a misunderstanding here.
> I'm using 1.3 alpha. This is a compiled version using AUR from Arch Linux https://aur.archlinux.org/packages/lmms-git/.
> 
> The issue described here, still there.

Ok, so its a Master 1.3 issue, and **not** in off. release.
Over and out..

**Comment by rdrpenguin04** [comment]
> Ok, so its a Master 1.3 issue, and **not** in off. release.
> Over and out..
@musikBear Feel free to shut it; this is a legitimate problem, especially if it's on the latest source-built versions.

Anyways, no idea what the status of this is. Could we get a report from whoever is in charge of this now?

**Comment by musikBear** [comment]
Nonono should not be closed!
Its just that the _version info_ was missing, because this was added to a very old issue, so the form used for reporting bugs was not available -No, good report!

**Comment by allejok96** [comment]
There are multiple examples mentioned here, one is fixed and two is pending.

- [ ] Undo insert bar #6347
- [x] Undo track color change #6165
- [ ] Undo copy paste of multiple clips #5920

Note that undo system is still full of holes and needs to be reworked. That's a topic for another issue.

**Comment by Rossmaxx** [comment]
Re milestone ing till someone takes over #6410
